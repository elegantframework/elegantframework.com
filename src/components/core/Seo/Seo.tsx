import { DefaultSeo } from 'next-seo';
import Favicon from './Favicon/Favicon';
import MSApplicationTile from './Meta/MSApplicationTile';

interface Props {
    /**
     * The title of the app.
     */
    title?: string;
    /**
     * The description of the app.
     */
    description?: string;
    /**
     * The theme color.
     */
    themeColor?: string;
    /**
     * Is this the base placeholder seo?
     */
    base?: boolean;
};

/**
 * A simple SEO head component for rendering all of the perfect seo meta data you could need.
 * @returns All of the seo meta data needed to inject into the html head component.
 */
const Seo = ({
    title = "",
    description = "",
    themeColor = "#ffffff",
    base = false
}: Props) => {
    if(!base){

    }

    return(
        <DefaultSeo 
            title={title}
            description={description}
            additionalLinkTags={
                Favicon({
                    directory: "/favicons"
                })
            }
            additionalMetaTags={
                MSApplicationTile({
                    directory: "/favicons",
                    color: themeColor
                })
            }
            themeColor={themeColor}
        />
    );
};

export default Seo;