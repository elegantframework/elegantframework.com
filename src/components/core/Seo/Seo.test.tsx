import React from "react";
import renderer from 'react-test-renderer';
import Seo from "./Seo";

describe('Home', () => {
  it('renders the seo component properly', () => {
    const seo = renderer
    .create(<Seo />)
    .toJSON();

    expect(seo).toMatchSnapshot();
  })
})