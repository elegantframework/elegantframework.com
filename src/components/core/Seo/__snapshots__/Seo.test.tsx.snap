// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders the seo component properly 1`] = `null`;
